Chronological.Tests/FilterExpressionTests.cs

```diff
@@ -25,6 +25,18 @@ public void FilterPredicateTests(Expression<Func<TestType1, bool>> predicate, st
 
             Assert.Equal(expected, predicateString);
         }
+
+        [Theory]
+        [MemberData(nameof(FilterPredicateTestDataProvider.TestBreakingCase), MemberType = typeof(FilterPredicateTestDataProvider))]
+        public void FilterPredicateBreakingTests(Expression<Func<TestType1, bool>> predicate, string expected)
```

Comment:
We don't need a special method for breaking tests just add it to the existing IEnumerable of test cases for predicates

Reply:
I just didn't found how to set a value from a variable (or anonym object) inside the existing test cases. The only way I see is to populate testCases from the Property itself : 
```csharp
        public static IEnumerable<object[]> TestCases
        {
            get
            {
                var firstObj = new { Prop1 = "Val1", Prop2 = "TestValue" };
                var secondObj = new { PropA = firstObj, PropB = "Test" };
                (Expression<Func<TestType1, bool>> Expr, string Expect) testCase = (x => x.DataType == secondObj.PropA.Prop2, "([data.type] = 'TestValue')");
                _testCases.ToList().Add(testCase);

                foreach (var (expr, expect) in _testCases)
                {
                    yield return new object[] { expr, expect };
                }
            }
        }
```

But I found it more messy that the way it's already done :)

Reply:
I think something like this is cleaner, it saves having two tests and two enumerables:

```cs

        public static IEnumerable<object[]> TestCases
        {
            get
            {
                foreach (var (expr, expect) in _testCases)
                {
                    yield return new object[] { expr, expect };
                }
                (var sc1Expr, var sc1Expect) = SpecialCase1();
                yield return new object[] { sc1Expr, sc1Expect };
            }
        }

        static (Expression<Func<TestType1, bool>> Expr, string Expect) SpecialCase1()
        {
            var firstObj = new { Prop1 = "Val1", Prop2 = "TestValue" };
            var secondObj = new { PropA = firstObj, PropB = "Test" };
            return (x => x.DataType == secondObj.PropA.Prop2, "([data.type] = 'TestValue')");
        }
```